packages/dapi/lib/grpcServer/handlers/blockheaders-stream/getHistoricalBlockHeadersIteratorFactory.js

```diff
@@ -0,0 +1,56 @@
+const { BlockHeader } = require('@dashevo/dashcore-lib');
+
+const MAX_HEADERS_PER_REQUEST = 500;
+
+/**
+ * @param {number} batchIndex
+ * @param {number} numberOfBatches
+ * @param {number} totalCount
+ * @return {number}
+ */
+function getBlocksToScan(batchIndex, numberOfBatches, totalCount) {
+  const isLastBatch = batchIndex + 1 === numberOfBatches;
+  return isLastBatch
+    ? totalCount - batchIndex * MAX_HEADERS_PER_REQUEST
+    : MAX_HEADERS_PER_REQUEST;
+}
+
+/**
+ * @param {CoreRpcClient} coreRpcApi
+ * @return {getHistoricalBlockHeadersIterator}
+ */
+function getHistoricalBlockHeadersIteratorFactory(coreRpcApi) {
+  /**
+   * @typedef getHistoricalBlockHeadersIterator
+   * @param fromBlockHash
+   * @param count
+   * @return {AsyncIterableIterator<BlockHeader[]>}
+   */
+  async function* getHistoricalBlockHeadersIterator(
+    fromBlockHash,
+    count,
+  ) {
+    const fromBlock = await coreRpcApi.getBlock(fromBlockHash);
```

Comment:
Better use https://dashcore.readme.io/docs/core-api-ref-remote-procedure-calls-blockchain#getblockstats and ask only for height

Reply:
I tried, but unfortunately, `dashd-rpc` does not expose this method yet.
Added TODO

Reply:
Sounds good!